### ISSUE TITLE
Generic AuthException is thrown when the device does not have chrome installed

### ISSUE THREAD

**Issue by PawlikMichal25** [aws-amplify/amplify-flutter#587]
**Describe the bug**
The plugin uses CustomTabsManagerActivity which won't be available if Chrome is uninstalled on a device.

**To Reproduce**
1. Uninstall Chrome, you can do it on any Pixel device
2. Try calling `signInWithWebUI()`
4. ~~App will crash, happens every time~~ A generic AuthException is thrown

**Expected behavior**
~~At least nothing should happen instead of crash, but special error for this would be better.~~
A more specific exception should be thrown

**Platform**
[X] Android

**Stacktrace**
```
Fatal Exception: java.lang.RuntimeException: Unable to resume activity {MYPACKAGENAME/com.amazonaws.mobileconnectors.cognitoauth.activities.CustomTabsManagerActivity}: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=MYURL/... pkg=com.android.chrome (has extras) }
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3785)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3817)
       at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:193)
       at android.app.ActivityThread.main(ActivityThread.java:6680)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by haverchuck** [comment]
@BaranMichal25 Do you have a particular Android browser you would try to target instead of chrome?

**Comment by Sunsiha** [comment]
any update on this?

**Comment by Jordan-Nelson** [comment]
I believe there were some changes in Amplify-Android (which Amplify-Flutter depends on when targeting Android) to resolve this. I just tested this on a Pixel 4 with Chrome disabled with Amplify-Flutter 0.4.5. I do not see the app crash. I see a generic `AuthException`.

As the original issue suggested, a more specific Exception would be better. I am going to update the issue title to reflect that.

**Comment by Jordan-Nelson** [comment]
Hello - We have recently released a new developer preview version of the Auth category ([amplify_flutter v1.0.0-next.0](https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.0)). This new version is a full re-write of the auth category in dart, and it supports Mobile, Desktop, and Web. This issue is addressed in this new version. You will now see a `UrlLauncherException` if the user has no supported browsers installed.

You can read more about the developer preview release in the [blog post](https://aws.amazon.com/blogs/mobile/amplify-flutter-auth-web-and-desktop/) and [docs](https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/). I will update this issue again once this new version is out of developer preview.

Let me know if you have any questions!

**Comment by Jordan-Nelson** [comment]
This issue has been addressed in v1.0.0 of Amplify Flutter, which is now stable. This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).
